Set enableScriptExecution to false by default

kachkaev opened the pull request:
Disabling code chunk execution by default wil improve the security of MPE users.

Not all of us are using MPE for code chunk execution and some of us are not even aware of this feature. However, because it is on by default, there is a higher chance that some hackers may social-engineer a MPE user and, say, steal their private data by making them them open a markdown file with the malicious code. Forcing everyone who wants to enjoy live code chunks to set `enableScriptExecution` to `true` increases their awareness about the potential security issues and thus makes them less vulnerable to attacks.

A new look of preferences pane:

<img width="572" alt="screen shot 2018-04-04 at 15 03 53" src="https://user-images.githubusercontent.com/608862/38312349-72bd00f8-3819-11e8-95d8-ca59e11e2aa2.png">

VSCode extension PR: https://github.com/shd101wyy/vscode-markdown-preview-enhanced/pull/96

kachkaev commented:
> Replying to https://github.com/shd101wyy/markdown-preview-enhanced/issues/838

Oh I see – TOC is also a code chunk and so it'd be disabled too if we set `enableScriptExecution` to `false`. But what if we move all the TOC-related stuff to a new renderEnhancer and even simplify the attributes needed? That could be a useful thing to do given that a TOC does not create any vulnerabilities. The only thing to worry about is attribute backwards-compatibility, which we should keep in mind.

I could work on a PR that does it, but I'll also be happy if you do it because then I'll be able to finish the [litvis](https://github.com/gicentre/litvis)-related npm refactoring quicker.

kachkaev commented:
What is great is that TOC is already [split from code chunks in the docs](https://shd101wyy.github.io/markdown-preview-enhanced/#/toc), so we don't complicate or change things for users. It is only the implementation detail that gets amended (unless we decide to simplify TOC attributes, of course).

shd101wyy commented:
Sure, I will do it this weekend. Thanks you!

On Wed, Apr 4, 2018 at 11:25 AM, Alexander Kachkaev <
notifications@github.com> wrote:

> What is great is that TOC is already split from code chunks in the docs
> <https://shd101wyy.github.io/markdown-preview-enhanced/#/toc>, so we
> don't complicate or change things for users. It is only the implementation
> detail that gets amended (unless we decide to simplify TOC attributes, of
> course).
>
> —
> You are receiving this because your review was requested.
> Reply to this email directly, view it on GitHub
> <https://github.com/shd101wyy/markdown-preview-enhanced/pull/837#issuecomment-378661534>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AB0gf8r3RAovss5F1WJWhGqNNXw6hy7gks5tlPQUgaJpZM4TG21d>
> .
>



-- 

Yiyi Wang

Full Stack Engineer

+1-217-649-9936 <https://github.com/shd101wyy>
<https://www.facebook.com/yiyi.wang.336>
<https://www.linkedin.com/in/yiyi-wang-60416380/>

Hangouts: shd101wyy

 | Skype: shd101wyy
<https://runtimeverification.com/>

Runtime Verification, Inc.

102 E Main St #500, Urbana, IL 61801
<https://maps.google.com/?q=102+E+Main+St+%23500,+Urbana,+IL+61801&entry=gmail&source=g>

<https://github.com/runtimeverification>
<https://www.facebook.com/RuntimeVerificationInc>
<https://www.linkedin.com/company/3142238/> <https://twitter.com/rv_inc>

shd101wyy commented:
@kachkaev I just realized that the TOC case is already considered in [fenced-code-chunks.ts line 192](https://github.com/shd101wyy/mume/blob/312e9136daef2d3c124fb643abca8d3035b2b838/src/render-enhancers/fenced-code-chunks.ts#L192) and [fenced-code-chunks.ts line 207](https://github.com/shd101wyy/mume/blob/312e9136daef2d3c124fb643abca8d3035b2b838/src/render-enhancers/fenced-code-chunks.ts#L207) so I think this pull request could be safely merged. Thank you.

kachkaev commented:
The conflict was because of https://github.com/shd101wyy/markdown-preview-enhanced/pull/839 – we also changed `config-schema.js.map` there. I rebased this PR on the latest master, so no more conflicts.

IMO sourcemaps are useful as they help generate stack traces that are easier for us to understand. That said, I hope plugin users won't see many of them 😅
